src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,60 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int HOURS_INDEX = 2;
+    private static final int SALARY_PER_HOUR_INDEX = 3;
+    private static final int ONE_DAY = 1;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateStart = LocalDate.parse(dateFrom, formatter);
+        LocalDate dateEnd = LocalDate.parse(dateTo, formatter);
+        LocalDate comparisonDate;
+        int[] salaryOfNames = new int[names.length];
+        final int numberOfWorkers = names.length;
+        final int lastIndexOfData = data.length - 1;
+        int countsOfValideDays = 0;
+        int nameNumber = 0;
+
+        for (int i = 0; i < data.length; i++) {
+            String[] oneLineArray = data[i].split(" ");
+            comparisonDate = LocalDate.parse(oneLineArray[DATE_INDEX], formatter);
+            if (comparisonDate.isAfter(dateStart.minusDays(ONE_DAY))
+                    && comparisonDate.isBefore(dateEnd.plusDays(ONE_DAY))) {
+                countsOfValideDays++;
+                if (oneLineArray[NAME_INDEX].equals(names[nameNumber])) {
+                    salaryOfNames[nameNumber] += (Integer.parseInt(oneLineArray[HOURS_INDEX]))
+                            * (Integer.parseInt(oneLineArray[SALARY_PER_HOUR_INDEX]));
+
+                }
+
+            }
+            if (i == lastIndexOfData && countsOfValideDays != 0) {
+                nameNumber++;
+                countsOfValideDays = 0;
+                i = 0;
```

Comment:
This is tricky hack to have inner loop using one `for` statement))) Do not do it

See how it can be implemented
![image](https://github.com/mate-academy/jv-salary-info/assets/20326858/f832a8f8-cd6b-43dd-8f8a-ded5b0baf26d)

Reply:
At this stage, this structure is too complicated for me. What's the problem if the code solves the problem and it's clean?

Reply:
I can't figure out what's the problem with my code? Does he solve the problem? decides. Does it take into account the condition of not using two loops? Takes it into account. Why did you decide that your decision was the only right one? As far as it seems to me, it does not exist and there are only optimal and non-optimal solutions. If you think my solution is not optimal, please justify this and not just say that it is not suitable.

Reply:
Several aspects why my code is more optimal:

Using Explicit Constants:

The second code uses named constants for indices and other values, which makes the code more readable and maintainable.
Loop with Explicit Indexes:

Using a single loop with explicit indexes makes the code more compact and makes it easier to understand the order of operations.
Data Processing in One Cycle:

The second code first processes all the data and then generates a report for all employees. This can be more efficient in certain situations because it reduces the number of passes through the data.
Explicit Index Management:

Explicitly managing indexes and counters in a loop makes the code clearer and more predictable.

Reply:
Several aspects why my code is more optimal:

Using Explicit Constants:
I use named constants for indices and other values, which makes the code more readable and maintainable.
Loop with Explicit Indexes:

Using a single loop with explicit indexes makes the code more compact and makes it easier to understand the order of operations. Nested loops are evil, and I didn't say that.
Data Processing in One Cycle:

I first process all the data and then generate a report for all employees. This can be more efficient in certain situations because it reduces the number of passes through the data.

Explicit Index Management:
Explicitly managing indexes and counters in a loop makes the code clearer and more predictable.

Reply:
why your solution is bad:
1 Not trivial usage of the for loop : Very hard to understand
2 Hard to debug in case of errors (because of index that reset) 
3 You dont understand why inner loop is bad -> because of increasing time complexity, e.g. O(n2) with 2 loops, O(n3) with 3 loops and so on. Your solution dont have win by time as well. 
 4 You cant easily reproduce code in your head in a few seconds. 
 
 If you want to talk more about this, come to QnA.

Reply:
No. its enough. I will change. Thanks you :)